packages/cli/src/commands/start.ts

```diff
@@ -288,6 +288,11 @@ export class Start extends BaseCommand {
 
 		await this.initPruning();
 
+		if (config.getEnv('executions.mode') === 'regular') {
+			// cancel any lingering enqueued executions due to instance crash
+			await Container.get(ExecutionRepository).cancelAllNewExecutions();
+		}
+
```

Comment:
why are we cancelling these executions? could we not re-enqueue them?

During shutdown, we should cancel any executions that have a `responsePromise`. All other executions with `status=new` could be resumed at the next start-up.

Reply:
This was discussed and descoped from this first iteration, see [here](https://www.notion.so/n8n/Concurrency-limits-for-main-mode-8b642df5e98a479cadd1cfc4950d27a1?pvs=4#041f3c1d9bfb4d3f83ab68df6a0ddd98). Let me know if you feel strongly about this and I'll bring it up again.

Reply:
I think we need resume-at-startup for new executions before we can start setting concurrency limits on the cloud. otherwise some of our heavy-usage customers will start seeing executions failing even before they start, since many of those instances crash and restart often.
IMO this needs to be done before the paywall stuff.

Reply:
https://github.com/n8n-io/n8n/pull/9658